### ISSUE TITLE
invalid preprocessor command 'warning'

### ISSUE THREAD

**Issue by mdavidsaver** [numpy/numpy#27224]
https://github.com/numpy/numpy/blob/282c79ebc9b61ebc5f0755736ec953b8d954922c/numpy/_core/include/numpy/numpyconfig.h#L133

Results in an compile error `invalid preprocessor command 'warning'
` when building against numpy 2.0.1 with MSVC from the windows-2019 GHA image.

```
  C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\numpy\_core\include\numpy\numpyconfig.h(132): fatal error C1021: invalid preprocessor command 'warning'
```

Another usage in `npy_1_7_deprecated_api.h` is conditional on `defined(_WIN32)`, which avoids the issue.

https://github.com/numpy/numpy/blob/282c79ebc9b61ebc5f0755736ec953b8d954922c/numpy/_core/include/numpy/npy_1_7_deprecated_api.h#L10-L17

**Comment by mattip** [comment]
TL;DR: we should fix this. Want to submit a PR?

I was surprised to see github actions [still supports](https://github.com/actions/runner-images?tab=readme-ov-file#available-images) windows-2019 which has [visual studio 2019](https://github.com/actions/runner-images/blob/main/images/windows/Windows2019-Readme.md#visual-studio-enterprise-2019). Apparently Microsoft has a [10 year support commitment](https://learn.microsoft.com/en-us/visualstudio/productinfo/vs-servicing#older-versions-of-visual-studio), so there is still a lot of time before that compiler goes EOL.

**Comment by mdavidsaver** [comment]
I have created https://github.com/numpy/numpy/pull/27235 with the simple fix.

Some notes on C/C++ pre-processor support.  Apparently the `#warning` directive becomes standard with [C23](https://en.cppreference.com/w/c/preprocessor/error).  MSVC <= 2022 does not claim support (see the line [`P2437R1 Support for #warning`](https://learn.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-170)).  I have used `#warning` occasional for many years, and think of it like a [GNU extension](https://gcc.gnu.org/onlinedocs/cpp/Diagnostics.html#index-_0023warning).  However, I can find no reference which states that it originates with GCC.  Maybe there is a longer *NIX history?
